src/v/cluster/security_frontend.cc

```diff
@@ -339,4 +339,56 @@ security_frontend::get_bootstrap_user_creds_from_env() {
       std::in_place, std::move(username), std::move(credentials));
 }
 
+ss::future<result<model::offset>> security_frontend::get_leader_last_applied(
+  model::node_id leader, model::timeout_clock::duration timeout) {
+    return _connections.local()
+      .with_node_client<cluster::controller_client_protocol>(
+        _self,
+        ss::this_shard_id(),
+        leader,
+        timeout,
+        [timeout](controller_client_protocol cp) mutable {
+            return cp.get_controller_last_applied(
+              controller_last_applied_request{},
+              rpc::client_opts(model::timeout_clock::now() + timeout));
+        })
+      .then(&rpc::get_ctx_data<controller_last_applied_reply>)
+      .then([](result<controller_last_applied_reply> r) {
+          if (r.has_error()) {
+              return result<model::offset>(r.error());
+          } else if (r.value().result != errc::success) {
+              return result<model::offset>(r.value().result);
+          }
+          return result<model::offset>(r.value().last_applied);
+      });
+}
+
+ss::future<std::error_code> security_frontend::wait_until_caughtup_with_leader(
+  model::timeout_clock::duration timeout) {
+    auto leader = _leaders.local().get_leader(model::controller_ntp);
+    if (!leader) {
+        co_return make_error_code(errc::no_leader_controller);
+    }
+    if (leader == _self) {
+        co_return make_error_code(errc::success);
+    }
```

Comment:
In this case don't need to wait until stm catches up? eg: acl applied but the stm is not yet aware of it. I think we ultimately care if the stm applied it (so that we can query the state) and not whether it is raft committed.

Reply:
This PR has followers wait until the leaders last applied, whatever the value of that is `now`. If this logic is applied to the leader, the last applied has already been applied, so no need to wait. So the semantics kind of match now between how the command works when handled by a follower or leader.